elyra/metadata/schemas/component-registry.json

```diff
@@ -0,0 +1,75 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Component Registry",
+  "name": "component-registry",
+  "display_name": "Components",
+  "namespace": "component-registries",
+  "uihints": {
+    "title": "Components",
+    "icon": "",
+    "reference_url": ""
+  },
+  "properties": {
+    "schema_name": {
+      "title": "Schema Name",
+      "description": "The schema associated with this instance",
+      "type": "string",
+      "pattern": "^[a-z][a-z0-9-_]*[a-z0-9]$",
+      "minLength": 1
+    },
+    "display_name": {
+      "title": "Display Name",
+      "description": "Display name of this Component Registry",
+      "type": "string",
+      "minLength": 1
+    },
+    "metadata": {
+      "description": "Additional data specific to this metadata",
+      "type": "object",
+      "properties": {
+        "description": {
+          "title": "Description",
+          "description": "Description of this Component Registry",
+          "type": "string"
+        },
+        "runtime": {
+          "title": "Runtime",
+          "description": "The runtime associated with this Component Registry",
+          "type": "string"
+        },
+        "type": {
+          "title": "Location Type",
+          "description": "The type of location for this registry",
+          "type": "string",
+          "enum": ["URL", "Filename", "Directory"],
```

Comment:
Are we supporting GitHub?

Reply:
I know we want to eventually. I think in the issue, it says that we'll set the groundwork with directory, filename, and url types for now, and add GitHub support after. If it's trivial to support, I definitely think we should try to get it into this PR. But if not and we want to get this into 3.1.0, I can do an immediate follow up to add GitHub as well